### ISSUE TITLE
[MJAVADOC-376] please add property 'maven.javadoc.classifier'

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#682]
**[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus)** opened **[MJAVADOC-376](https://issues.apache.org/jira/browse/MJAVADOC-376?redirect=false)** and commented

We need an easier way to customize the name of the generated javadoc jar file.

* For maven-jar-plugin, there's a `classifier` attribute.
* For maven-source-plugin, it's even better, there's a `maven.source.classifier` property

But for maven-javadoc-plugin, there's no chance to customize, the classifier is hardcoded to "javadoc", making it very hard to customize the file name of the generated javadoc jar, which is needed when a classifier is needed to differentiate an artifact for multiple jdk vesions.

By adding support for the `maven.javadoc.classifier` property, it would be much more straightforward and consistent.

Thanks

---
No further details from [MJAVADOC-376](https://issues.apache.org/jira/browse/MJAVADOC-376?redirect=false)

**Comment by jira-importer** [comment]
**[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus)** commented

pull request sent on github, see https://github.com/apache/maven-plugins/pull/14

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

merged http://svn.apache.org/r1525844
Thanks!

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

I made some small adjustments ( http://svn.apache.org/r1525961 ) and added a similar parameter for the test-javadoc ( http://svn.apache.org/r1525968 )

**Comment by jira-importer** [comment]
**[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus)** commented

May I ask when would the next version get release?

And, what about updating the snapshot version in maven central so that I can start to use it?

Thank you

**Comment by jira-importer** [comment]
**[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus)** commented

Robert, thank you for the complements! I know I only did the minimum :-)

BTW, are we thinking about how to unify all these properties? I think we can have just one classifier property that applies to all kind of generated files, e.g. in my case to have the javadoc jar for jdk6 named properly, we need to set maven.javadoc.classifier to 'javadoc-jdk6', but shouldn't the 'javadoc' part already part of the file? Similarly

'sources' should be a builtin for sources jar filenames
'test-sources' should be a builtin for test sources jar filenames
'javadoc' should be a builtin for javadoc jar filenames
'test-javadoc' should be a builtin for test javadoc jar filenames
'test' should be a builtin for binary test jar filenames

this way, we just need to specify one and only one classifier 'jdk6' to have this classifier attached to all the names of the generated files.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

@[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus), the SNAPSHOT can be found at https://repository.apache.org/content/repositories/snapshots/. SNAPSHOTs will never end up in Maven Central.

I'm not sure if those properties can be unified, it all depends if the there's already a property available for every classifier. We don't want to break backwards compatibility.
Using `"jdk6"` as the classifier is confusing, because it's not the classifier you specify as a dependency.

Actually, I'm curious about your convention, because it seems like the same project also produces jdk4, jdk5 and jdk7, but all with different sources?

**Comment by jira-importer** [comment]
**[ryenus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ryenus)** commented

Right, such change would break backward compatibility, making it a simple thing but hard to make :-(

For my project I initial wrote it for java 7 and use new language features like diamond operator, multi-catch, try with resource, then I had to backport it to java 6 because not everyone can immediately upgrade to java 7. Therefore the source code must be changed by removing all those java 7 only stuff. Oh, there's all some some NIO classes new in java 7, like the Path class etc I had to remove as well.
